README.md

```diff
@@ -227,13 +279,19 @@ maxmcode=40960  -- in KB
 
 ### String hashing
 
-This optimization only applies to Intel CPUs supporting the SSE 4.2 instruction
-sets. For such CPUs, and when this branch is compiled with `-msse4.2`, the
-string hashing function used for strings interning will be based on an
-optimized crc32 implementation (see `lj_str_new()`).
+This optimizations modifies the string hashing algorithm to use a CRC32-based
+variant. This variant still provides constant-time hashing complexity (`O(n)`)
+but makes hash collision attacks harder for strings up to 127 bytes of size
+(see `lj_str_new()`).
+
+This optimization is only available for x64 and ARM64 architectures, and will
+be enabled if:
+
+1. LuaJIT is compiled with `-msse4.2` on x64, or `-march=armv8-a+crc` on ARM64.
```

Comment:
This PR does not enable this feature by default, at least not yet. I am not sure if it should, or if OpenResty's `configure` script should be responsible for enabling this feature, which I believe to be more appropriate.